app/controllers/declaration/CarrierDetailsPageController.scala

```diff
@@ -55,9 +57,16 @@ class CarrierDetailsPageController @Inject()(
       .fold(
         (formWithErrors: Form[CarrierDetails]) => Future.successful(BadRequest(carrierDetailsPage(formWithErrors))),
         form =>
-          customsCacheService.cache[CarrierDetails](cacheId, CarrierDetails.id, form).map { _ =>
-            Redirect(controllers.declaration.routes.DeclarationAdditionalActorsController.displayForm())
-        }
+          for {
```

Comment:
If you pull the for comprehension into the `updateCache` method (see LocationController) it means less refactoring when we switch the retrieves over

Reply:
This has been addressed!